frontend/src/pages/activityPage.tsx

```diff
@@ -43,124 +55,185 @@ export default function ActivitiesPage() {
-  }, []);
+    if (!currLesson) return;
+    setAnswers(Array(currLesson.activities.length).fill(undefined));
+    setCurrentIndex(0);
+  }, [currLesson]);
 
-  const lessonStatuses = getLessonStatuses(units.flatMap((unit) => unit.lessons));
+  const currentQuestion = currLesson?.activities[currentIndex];
+  const currentAnswer = currentQuestion ? (answers[currentIndex] ?? "") : "";
 
-  // keep track of the index to retrieve the correct status
-  let statusIndex = 0;
+  const handleAnswer = (answer: string) => {
+    if (!currLesson) return;
+    const updatedAnswers = [...answers];
+    updatedAnswers[currentIndex] = answer;
+    setAnswers(updatedAnswers);
+  };
+
+  const handleNext = () => {
+    if (!currLesson) return;
+    if (currentIndex < currLesson.activities.length - 1) {
+      setCurrentIndex((prev) => prev + 1);
+    } else {
+      void handleComplete();
+    }
+  };
+
+  const handleBack = () => {
+    if (currentIndex > 0) setCurrentIndex((prev) => prev - 1);
+  };
+
+  const handleComplete = async () => {
+    if (!mongoUser?._id || !currLesson?._id) {
+      alert("User or activity not found.");
+      return;
+    }
+
+    try {
+      const res = await fetch(
+        `${env.EXPO_PUBLIC_BACKEND_URI}/users/${mongoUser.uid}/completed/${String(currLesson._id)}`,
+        {
+          method: "PUT",
+          headers: { "Content-Type": "application/json" },
+        },
+      );
+
+      if (res.ok) {
+        alert("Activity completed!");
+        await refreshMongoUser();
+        setCurrLesson(null);
+      } else {
+        alert("Failed to update activity progress.");
+      }
+    } catch (err: unknown) {
+      console.error(err);
+      alert("Error updating activity progress.");
+    }
+  };
 
   return (
-    <View style={styles.container}>
-      {/* Header */}
-      <View style={styles.header}>
-        <TouchableOpacity
-          onPress={() => {
-            router.back();
-          }}
-        >
-          <Ionicons name="arrow-back-outline" size={24} color="gray" />
-        </TouchableOpacity>
-        <Text style={styles.headerTitle}>Activities</Text>
-      </View>
-
-      {/* Content */}
-      <ScrollView contentContainerStyle={styles.scrollContainer}>
-        {units.map((unit, sectionIndex) => {
-          return (
+    <SafeAreaView style={styles.container}>
+      {!currLesson ? (
+        // Activities list view
+        <ScrollView contentContainerStyle={styles.scrollContainer}>
+          <View style={styles.header}>
+            <TouchableOpacity
+              onPress={() => {
+                router.back();
+              }}
+            >
+              <Ionicons name="arrow-back-outline" size={24} color="gray" />
+            </TouchableOpacity>
+
+            <Text style={styles.headerTitle}>Activities</Text>
+          </View>
+
+          {units.map((unit, sectionIndex) => (
             <View key={unit._id} style={styles.sectionContainer}>
               <SectionButton
```

Comment:
Can you modify `SectionButton` to not display the hard-coded `Header 1`, `Header 2`, etc. dropdowns? I don't think we need dropdowns at all but lmk what you think